### PR TITLE
Java2Swift: Account for covariant overrides so we don't have duplicates

### PR DESCRIPTION
**Pull request by DougGregor**
When we see methods with the same name and parameter types but different
result types, check whether the result types are related and only keep
the more specialized method.

### CODE REVIEW

File: Sources/JavaKitCollection/generated/ArrayDeque.swift [hunk 1]
```diff
@@ -25,9 +25,6 @@ public struct ArrayDeque<E: AnyJavaObject> {
   @JavaMethod
   public func clone() -> ArrayDeque<JavaObject>?
 
-  @JavaMethod
-  public func clone() throws -> JavaObject?
-
```
Comment (DougGregor) [inline Sources/JavaKitCollection/generated/ArrayDeque.swift:30]: This method is subsumed by the covariant method above it, hence its removal.

**Comment by DougGregor** [comment]
It's a little hard to test the effect of this pull request in isolation, because we don't have a good way to test "does not emit this code", but I've marked where in the generated sources we drop a method because it's been subsumed by a covariant override.

**Review by ktoso — APPROVED** [review]
Looks good, nice to capture it in a "collector" 👍

File: Sources/JavaKitCollection/generated/ArrayDeque.swift (hunk 1 above)
Comment (ktoso) [reply Sources/JavaKitCollection/generated/ArrayDeque.swift:30]: Gotcha, makes sense +1

**Pull request merged**